src/NServiceBus.Core/Persistence/PersistenceDefinition.cs

```diff
@@ -68,6 +68,14 @@ public bool HasSupportFor(Storage storage)
             return storageToActionMap.ContainsKey(typeof(T));
         }
 
+        /// <summary>
+        /// True if supplied storage is supported
+        /// </summary>
+        public bool HasSupportFor(Type storageType) 
```

Comment:
+1

On Mon, Feb 9, 2015 at 1:58 PM, Szymon Pobiega <notifications@github.com>
wrote:

> In src/NServiceBus.Core/Persistence/PersistenceDefinition.cs
> <https://github.com/Particular/NServiceBus/pull/2644#discussion_r24325941>
> :
>
> > @@ -68,6 +68,14 @@ public bool HasSupportFor(Storage storage)
> >              return storageToActionMap.ContainsKey(typeof(T));
> >          }
> >
> > +        /// <summary>
> > +        /// True if supplied storage is supported
> > +        /// </summary>
> > +        public bool HasSupportFor(Type storageType)
>
> I think if it is fine for a given persistence assembly to support a subset
> of features, it should be also possible to query what that subset is.
> Although it would not be an API that users use, it is a part of a public
> contract of a persistence lib.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2644/files#r24325941>.
>